compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -2877,7 +2877,13 @@ public void visit(BLangSimpleVarRef varRefExpr, AnalyzerData data) {
                 actualType = varSym.type;
                 markAndRegisterClosureVariable(symbol, varRefExpr.pos, data.env, data);
             } else if ((symbol.tag & SymTag.TYPE_DEF) == SymTag.TYPE_DEF) {
-                actualType = symbol.type.tag == TypeTags.TYPEDESC ? symbol.type : new BTypedescType(symbol.type, null);
+                if (symbol instanceof BTypeDefinitionSymbol) {
```

Comment:
instead of using `instanceOf`, We can use `symbol kind` to do this check.

Reply:
Updated